infrastructure/src/main/java/org/corfudb/infrastructure/log/CompactionPolicy.java

```diff
@@ -0,0 +1,38 @@
+package org.corfudb.infrastructure.log;
+
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+/**
+ * The interface of stream log compaction policy, which decides
+ * which log segments will be selected for compaction.
+ *
+ * Created by WenbinZhu on 5/22/19.
+ */
+public interface CompactionPolicy {
```

Comment:
CompactionPolicy -> ICompactionPolicy

Reply:
The `I` prefix is an old-fashion pattern and we're avoiding it in recent patches. Slava had a strong argument on this.